drivers/dma/dma_intel_adsp_hda.c

```diff
@@ -479,10 +480,12 @@ void intel_adsp_hda_dma_isr(void)
 	for (i = 0; i < ARRAY_SIZE(host_dev); i++) {
 		dma_ctx = (struct dma_context *)host_dev[i]->data;
 		cfg = host_dev[i]->config;
-
-		for (j = 0; j < dma_ctx->dma_channels; j++) {
-			if (!atomic_test_bit(dma_ctx->atomic, j))
+		enabled_chs = atomic_get(dma_ctx->atomic);
```

Comment:
this suggests that the `.atomic` field cannot change while we're in the ISR? Also not by a different core?

Reply:
@lyakh It can change, but it's harmless for this loop as either we do one unnecessary check (for a channel that is already disabled by another core), or we have an additional ISR if it's enabled concurrently.